metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -298,6 +308,56 @@ def make_tableau_client(self, site: str) -> Server:
             ) from e
 
 
+class AccessRolesIngestionConfig(ConfigModel):
+    enable_workbooks: bool = Field(
+        default=True,
+        description="Whether or not to enable access role ingestion for workbooks. "
+        "Default: True",
+    )
+
+    group_name_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Filter for Tableau group names when ingesting access roles. "
+        "For example, you could filter for groups that include the term 'Consumer' in their name by adding '^.*Consumer$' to the allow list."
+        "By default, all groups will be added as roles. "
+        "You can both allow and deny groups based on their name using their name, or a Regex pattern. "
+        "Deny patterns always take precedence over allow patterns. ",
+    )
+
+    role_prefix: str = Field(
+        default="",
+        description="Prefix to append to the group name when generating the role name used in the request_url.",
+    )
+
+    group_substring_start: Optional[int] = Field(
+        default=None,
+        description="Use this property if you only need a substring of the group in Tableau to create the role used in the request_url.",
+    )
+
+    group_substring_end: Optional[int] = Field(
+        default=None,
+        description="Use this property if you only need a substring of the group in Tableau to create the role used in the request_url.",
+    )
+
+    role_description: str = Field(
+        default="",
+        description="Description of the role that will be displayed in the Access Management tab of Datahub.",
+    )
+
+    displayed_capabilities: Optional[List[str]] = Field(
+        default=None,
+        description="List of Tableau permission capabilities (e.g. Read, Write, Delete) that will be displayed as Access Type in the Access Management tab of Datahub. "
+        "If omitted, all allowed capabilities of the group will be displayed.",
+    )
+
+    request_url: str = Field(
+        default="",
+        description="Request URL that is added to the role for requesting access. "
```

Comment:
As per code it looks like $ROLE_NAME is required to substitute the role name in request_url.

`Request URL that is added to access role metadata. The request URL must include $ROLE_NAME, For example 'https://example-host.com/order/$ROLE_NAME'`

Reply:
No, `$ROLE_NAME` shouldn't be required. If it's not present it just won't be replaced. Or where do you see an issue?